### ISSUE TITLE
Zero Value Notes - Bug or Feature?

### ISSUE THREAD

**Issue by Edicksonjga** [zingolabs/zingolib#449]
I'm not sure if the report is here or in the Zingo Mobile repository, however, I'm leaving it here.

I have made several transactions in my test wallet testing the Send All and Shield option, I have restored the same wallet in YWallet and to my surprise:

Zingo is generating Notes with zero (0) value every time I Send everything from one pool to another, Blindo All balance, I send everything from one wallet to another. 😳

**Steps (Shield Balance):**
1. Send all available balance from the Orchard pool to the Transparent pool.
2. Shields the balance that is in the Transparent pool.
3. Restore the wallet in YWallet.
4. When viewing the notes, you can see that two zero value notes (0) and the note with the available balance have been generated.

**Using Zero Value Notes in Zingo**

These zero value notes do not seem to affect Zingo when making a transaction, this means that Zingo does not seem to take these notes.

**TX ID made from Zingo when the wallet had 9 Zero Value Notes:**

https://zcashblockexplorer.com/transactions/7b34775b8c0b573666ecd39f7e6e497b5fdf57f853904128df04b6d25cf81273

**Image of the notes in YWallet:**

![Screenshot_20230823-220604.jpg](https://github.com/zingolabs/zingolib/assets/37910384/539e14b7-4cdb-4d99-aaa2-e44bc63ae9b5)

**Image of the transaction in the browser** where it can be seen that the transaction was carried out with the minimum number of actions:

![Screenshot_20230824-001904.jpg](https://github.com/zingolabs/zingolib/assets/37910384/32697bb4-7d7b-4c29-97bc-51b9e044edc8)

**Using Zero Value Notes in YWallet**

YWallet does take the zero value notes that Zingo generates, this concludes in a transaction with more logical actions.

**TX ID made from YWallet:**

https://zcashblockexplorer.com/transactions/e9a3224261b76449f2db83948806d33f0ac71dd8e0bd671bb1ca9d64229b9719

**Image of the transaction in the browser** where it can be seen that the transaction was carried out with all the notes and therefore the number of shares:

![Screenshot_20230824-002858.jpg](https://github.com/zingolabs/zingolib/assets/37910384/099ebd22-f4d8-453e-9048-d94d718cb183)

**What is the problem?**

On Zingo, at the time of making a transaction, probably none, but when syncing, I guess the wallet needs to process all those zero-value notes that Zingo is creating.

But, assuming this does affect Zingo when making a transaction, two things could happen:

1. Before implementing the **ZIP 317**, the transaction will be done with many logical actions for a small commission which is as a result that the transaction is not validated and is reversed.
2. After implementing **ZIP 317**, the transaction will be done with many logical actions due to the number of zero value notes, resulting in a very high commission transaction.

Although Zingo ignores zero value notes when making a transaction, this entire explanation of how this could affect Zingo or other wallets is summed up in one question...

**Why and for what purpose is Zingo Wallet creating these zero value notes?** 🤔

**Comment by fluidvanadium** [comment]
this looks like a bug

**Comment by juanky201271** [comment]
I can see this in a old wallet... (a lot of dust notes...)
![image](https://github.com/zingolabs/zingolib/assets/43755989/50a343e1-7532-4476-ae43-c6ff4f497353)

**Comment by Edicksonjga** [comment]
> I can see this in a old wallet... (a lot of dust notes...)
> ![image](https://github.com/zingolabs/zingolib/assets/43755989/50a343e1-7532-4476-ae43-c6ff4f497353)
> 

Yes, for now it only affects the synchronization, but if it is not corrected, it could be a headache if it is supported by sl ZIP 317, although they could also continue to exclude those notes, but I do not think it is the best solution.
